src/main/java/com/minecolonies/coremod/inventory/CraftingGUIBuilding.java

```diff
@@ -0,0 +1,314 @@
+package com.minecolonies.coremod.inventory;
+
+import com.google.common.collect.ImmutableList;
+import com.minecolonies.api.util.ItemStackUtils;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.*;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.CraftingManager;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CraftingGUIBuilding extends Container
+{
+    /**
+     * Amount of rows in the player inventory.
+     */
+    private static final int PLAYER_INVENTORY_ROWS = 3;
+
+    /**
+     * Amount of columns in the player inventory.
+     */
+    private static final int PLAYER_INVENTORY_COLUMNS = 9;
+
+    /**
+     * Initial x-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_X_OFFSET = 8;
+
+    /**
+     * Initial y-offset of the inventory slot.
+     */
+    private static final int PLAYER_INVENTORY_INITIAL_Y_OFFSET = 84;
+
+    /**
+     * Each offset of the inventory slots.
+     */
+    private static final int INVENTORY_OFFSET_EACH = 18;
+
+    /**
+     * Initial y-offset of the inventory slots in the hotbar.
+     */
+    private static final int PLAYER_INVENTORY_HOTBAR_OFFSET = 142;
+
+    /**
+     * The x position of the crafting result slot position.
+     */
+    private static final int X_CRAFT_RESULT = 124;
+
+    /**
+     * The y position of the crafting result slot position.
+     */
+    private static final int Y_CRAFT_RESULT = 35;
+
+    /**
+     * The x offset of the crafting slot position.
+     */
+    private static final int X_OFFSET_CRAFTING = 30;
+
+    /**
+     * The y offset of the crafting slot position.
+     */
+    private static final int Y_OFFSET_CRAFTING = 17;
+
+    /**
+     * Amount of slots in the crafting window.
+     */
+    private static final int CRAFTING_SLOTS = 5;
+
+    /**
+     * Amount of slots per line in the GUI.
+     */
+    private static final int SLOTS_PER_LINE = 9;
+
+    /**
+     * Start slot of the player hotbar.
+     */
+    private static final int HOTBAR_START = 32;
+
+    /**
+     * Total amount of slots in the GUI.
+     */
+    private static final int TOTAL_SLOTS = 41;
+
+    /**
+     * The crafting matrix inventory (2x2).
+     */
+    private final InventoryCrafting craftMatrix = new InventoryCrafting(this, 2, 2);
+
+    /**
+     * The crafting result slot.
+     */
+    private final IInventory craftResult = new InventoryCraftResult();
+
+    /**
+     * The world object.
+     */
+    private final World worldObj;
+
+    public CraftingGUIBuilding(final InventoryPlayer playerInventory, final World worldIn)
+    {
+        super();
+        this.worldObj = worldIn;
+
+        this.addSlotToContainer(new SlotCrafting(playerInventory.player, this.craftMatrix, this.craftResult, 0, X_CRAFT_RESULT, Y_CRAFT_RESULT) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)